### ISSUE TITLE
Migration fail adding a new RealmObject with a PrimaryKey field

### ISSUE THREAD

**Issue by wonTonx** [realm/realm-java#2008]
I have a class below when I try to add it at a new version it fail and gave me a stacktace like this

‘‘                                                                                Caused by: io.realm.exceptions.RealmMigrationNeededException: Field 'actionId' does support null values in the existing Realm file. Remove @Required or @PrimaryKey from field 'actionId' or migrate using io.realm.internal.Table.convertColumnToNotNullable().
’’

and i using realm 0.87

```java
public class RecognizeAction extends RealmObject {

    @PrimaryKey
    private String actionId;

    public String getActionId() {
        return actionId;
    }

    public void setActionId(String actionId) {
        this.actionId = actionId;
    }

    private String picUrl;
    private String state;

    public String getPicUrl() {
        return picUrl;
    }

    public void setPicUrl(String picUrl) {
        this.picUrl = picUrl;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }
}
```

**Comment by beeender** [comment]
This should be fixed by #2003 
Before it released, you can use 
```
addField("actionId", String.class, FieldAttribute.PRIMARY_KEY, FieldAttribute.REQUIRED)
```
as a workaround.

Sorry for the inconvenience

**Comment by wonTonx** [comment]
this is work thankyou
